pkg/network/discovery.go

```diff
@@ -6,6 +6,7 @@ import (
 
 const (
 	maxPoolSize = 200
+	connRetries = 3
```

Comment:
nit: probably it will make sense to move it to config. But not now maybe.

Reply:
Maybe, but that's not a priority for now, certainly.